### PR TITLE
[linting] enable more rules for revive and address issues

### PR DESCRIPTION
**Pull request by hezhizhen**
## Summary

1. Enable more rules for revive.
2. Address existing issues.

## How was it tested?

devbox run lint

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -35,7 +35,18 @@ linters-settings:
...
+      - name: atomic
+      - name: bare-return
+      - name: bool-literal-in-expr
+      - name: cognitive-complexity
+        arguments:
+          - 36 # TODO: gradually reduce it
+      - name: datarace
+      - name: duplicated-imports
+      - name: early-return
+      - name: empty-lines
```
Comment (hezhizhen) [inline .golangci.yml:48]: @savil If you don't like it, let's remove this.

Comment (savil) [reply .golangci.yml:48]: yeah, not a huge fan, lets remove please....

**Review by savil — APPROVED** [review]
thanks!

File: .golangci.yml (hunk 1 above)
Comment (hezhizhen) [reply .golangci.yml]: Removed.

**Pull request merged**